lib/puppet/http/service/puppetserver.rb

```diff
@@ -26,13 +26,24 @@ def initialize(client, session, server, port)
   # @api private
   #
   def get_simple_status(ssl_context: nil)
-    response = @client.get(
-      with_base_url("/status/v1/simple/master"),
-      headers: add_puppet_headers({}),
-      options: {ssl_context: ssl_context}
-    )
+    request_path = "/status/v1/simple/master"
 
-    process_response(response)
+    begin
+      response = @client.get(
+        with_base_url(request_path),
+        headers: add_puppet_headers({}),
+        options: {ssl_context: ssl_context}
+      )
+
+      process_response(response)
+    rescue Puppet::HTTP::ResponseError => e
+      if e.response.code == 404 && e.response.url.path == "/status/v1/simple/master"
+        request_path = "/status/v1/simple/server"
+        retry
+      else
+        raise e
+      end
+    end
```

Comment:
I was wondering why we are not using `/status/v1/simple` here instead of master|server 
https://github.com/puppetlabs/puppetserver/blob/6.x/documentation/status-api/v1/simple.markdown

```
❯ curl -k https://lower-soldiery.delivery.puppetlabs.net:8140/status/v1/simple
running%
```

Reply:
without specifying a service the bare `simple` endpoint will return success only if _all_ services are running. All services are required for FOSS. However there are PE HA scenarios where the overall service is degraded and, though some services are in a failed state, as long as the master/server service is up we need to continue.